src/mimefactory.rs

```diff
@@ -1784,11 +1785,11 @@ mod tests {
             .chars()
             .all(|c| c.is_ascii_alphanumeric() || c == ' '));
 
-        let s = format!("{}", new_address_with_name(display_name, addr.to_string()));
+        let s = render_email_address(display_name, addr);
 
         println!("{s}");
 
-        assert_eq!(s, "=?utf-8?q?=C3=A4_space?= <x@y.org>");
+        assert_eq!(s, r#""=?utf-8?B?w6Qgc3BhY2U=?=" <x@y.org>"#);
```

Comment:
I propose bringing back a variant of these tests because I do think they were valuable, testing that encoding stays sane. I pushed a commit that does so, feel free to modify of course.

Note that the encoding is completely different now: `"=?utf-8?B?w6Qgc3BhY2U=?=" <x@y.org>` (also note that there are quotes around the name now). I tested that at least GMX and Gmail can handle the new encoding just fine.